src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -374,6 +376,26 @@ impl CloudHypervisorInner {
             cmd.args(["--seccomp", "false"]);
         }
 
+        let netns = self.netns.clone();
+        if self.netns.is_some() {
+            info!(
+                sl!(),
+                "set netns for vmm : {:?}",
+                self.netns.as_ref().unwrap()
+            );
+        }
+
+        unsafe {
+            let _pre = cmd.pre_exec(move || {
+                if let Some(netns_path) = &netns {
+                    let netns_fd = std::fs::File::open(netns_path);
+                    let _ = setns(netns_fd?.as_raw_fd(), CloneFlags::CLONE_NEWNET)
+                        .context("set netns failed");
+                }
+                Ok(())
```

Comment:
@amshinde I did a bit of reading on `pre_exec` but I'm not too familiar with it; is this `Ok(())` required here?

Reply:
Yep - `pre_exec()` accepts a function or closure that needs to return `Result<()>`.